core/storage/auth/gae_models.py

```diff
@@ -133,10 +147,13 @@ def get_by_auth_id(cls, provider_id, auth_id):
         """
 
         if provider_id == feconf.GAE_AUTH_PROVIDER_ID:
-            return cls.query(cls.gae_id == auth_id).get()
+            model = cls.query(cls.gae_id == auth_id).get()
         elif provider_id == feconf.FIREBASE_AUTH_PROVIDER_ID:
-            return cls.query(cls.firebase_auth_id == auth_id).get()
-        return None
+            model = cls.query(cls.firebase_auth_id == auth_id).get()
+        else:
+            return None
+        model = cast(Optional[UserAuthDetailsModel], model)
```

Comment:
Add newline above this one, since it's logically separate.

Also try not to redeclare variables; can you just return cast(...) directly?

Reply:
Done